runner_manager/models/backend.py

```diff
@@ -116,3 +116,60 @@ def configure_instance(self, runner: Runner) -> Instance:
             labels=self.labels,
             metadata=Metadata(items=items),
         )
+
+
+class AWSConfig(BackendConfig):
+    """Configuration for AWS backend."""
+
+    region: str = "us-west-2"
+    subnet_id: str
+
+
+class AWSInstanceConfig(InstanceConfig):
+    """Configuration for AWS backend instance."""
+
+    image: str = "ami-0735c191cf914754d"  # Ubuntu 22.04 for us-west-2
+    instance_type: str = "t3.micro"
+    subnet_id: Optional[str] = None
+    security_group_ids: Optional[List[str]] = []
+    max_count: int = 1
+    min_count: int = 1
+    user_data: Optional[str] = ""
+    block_device_mappings: Optional[List[Dict[str, str]]] = []
+    tags: Optional[Dict[str, str]] = None
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    def configure_instance(self, runner: Runner) -> Dict:
+        """Configure instance."""
+        instance_config = {
+            "ImageId": self.image,
+            "InstanceType": self.instance_type,
+            "SubnetId": self.subnet_id,
+            "SecurityGroupIds": self.security_group_ids,
+            "TagSpecifications": [
+                {
+                    "ResourceType": "instance",
+                    "Tags": [
+                        {
+                            "Key": "Name",
+                            "Value": runner.name,
+                        },
+                        {
+                            "Key": "runner-manager",
+                            "Value": runner.manager,
+                        },
+                    ],
+                }
+            ],
+            "UserData": self.user_data,
+            "MaxCount": self.max_count,
+            "MinCount": self.min_count,
+            "BlockDeviceMappings": self.block_device_mappings,
+        }
+        if self.tags:
+            instance_config["TagSpecifications"][0]["Tags"].extend(
+                [{"Key": key, "Value": value} for key, value in self.tags.items()]
+            )
```

Comment:
I would find it easier to process (and to read) if you were defining the tags before the whole dict, like:

```suggestion
        tags = {
            "Name": runner.name,
            "runner-manager": runner.manager,
        }
        tags.update(self.tags)
        instance_config = {
            "ImageId": self.image,
            "InstanceType": self.instance_type,
            "SubnetId": self.subnet_id,
            "SecurityGroupIds": self.security_group_ids,
            "TagSpecifications": [
                {
                    "ResourceType": "instance",
                    "Tags": [{"Key": key, "Value": value} for key, value in tags.items()]
                }
            ],
            "UserData": self.user_data,
            "MaxCount": self.max_count,
            "MinCount": self.min_count,
            "BlockDeviceMappings": self.block_device_mappings,
        }
```